app/views/questionnaires/view.html.erb

```diff
@@ -25,5 +25,5 @@
 <BR/>
 
     <BR/>
-    <%= link_to 'Edit', :action => 'edit', :id => @questionnaire %> |
+    <%= link_to 'Edit', :controller => :questionnaires, :action => :edit_quiz, :id => @questionnaire.id, :pid => @participant.id %>|
```

Comment:
This may not be correct since this view may also be called by questionnaire types.